src/validation.cpp

```diff
@@ -5128,3 +5130,28 @@ bool ChainstateManager::DetectSnapshotChainstate(CTxMemPool* mempool)
     this->InitializeChainstate(mempool, /*snapshot_blockhash*/ PathToSnapshotHash(*path));
     return true;
 }
+
+bool CChainState::TeardownSnapshotDatadir()
+{
+    if (!m_from_snapshot_blockhash) {
+        // Chainstate isn't based on a snapshot.
+        return false;
+    }
+    std::optional<fs::path> snapshot_datadir = FindSnapshotChainstateDatadir();
+
+    if (!snapshot_datadir) {
+        return false;
+    }
+
+    uint256 datadir_hash = PathToSnapshotHash(*snapshot_datadir);
+
+    if (datadir_hash != *m_from_snapshot_blockhash) {
+        LogPrintf("[snapshot] WARNING - unexpected blockhash for snapshot datadir (%s); expected %s\n",
```

Comment:
Same comment, unclear what a node operator should do with an invalid, name-different snapshot datadir.